### PR TITLE
Add `positive_score_impact` support for `rank_features`

### PR DESCRIPTION
**Pull request by Hronom**
### Description
Added support for `positive_score_impact` parameter for `rank_features`
 
### Issues Resolved
[[List any issues this PR will resolve]](https://github.com/opensearch-project/OpenSearch/issues/2724)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java
Comment (nknize) [inline modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java:79]:
```suggestion
           List.of(meta, positiveScoreImpact);
```

**Review by nknize — CHANGES_REQUESTED** [review]
Thanks for adding this! one little nitpick on creating the parameter list since we're at min java 11.

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java
Comment (reta) [inline modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java:89]: I see it being used by `final Parameter<Boolean> positiveScoreImpact`, the `RankFeaturesFieldType` could drop `positiveScoreImpactё` and use it from `RankFeaturesFieldType` since it has a direct reference to it (sorry, copied from https://github.com/opensearch-project/OpenSearch/pull/2726#issuecomment-1087702435)

**Comment by nknize** [comment]
Hi @Hronom! Thank you so much for submitting this! I was about to ask if you wanted to finish up the last few comments but saw you're in Ukraine. Our thoughts are with you and your family during these difficult times. 

We are happy to tidy up these last few comments for you and commit on your behalf.  Thank you so much for finding the time to contribute code to the project. We are extremely grateful!

**Comment by Hronom** [comment]
@nknize thank you! I will try to add yours feedbacks now

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java
Comment (Hronom) [reply modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java:89]: @reta I just follow same style like it's done in:
* `org.opensearch.index.mapper.RankFeatureFieldMapper`
* `org.opensearch.index.mapper.ScaledFloatFieldMapper`

They also have it in `Builder`s.

Also you can check that there `meta` on the same level and it's have same situation:
```
        private final Parameter<Boolean> positiveScoreImpact = Parameter.boolParam(
            "positive_score_impact",
            false,
            m -> ft(m).positiveScoreImpact,
            true
        );
        private final Parameter<Map<String, String>> meta = Parameter.metaParam();
```

I think it's better to be in the same style for now and if needed do refactoring for all places in separate PR.

However, I'm ok if maintainers of this repository do changes that is needed before merging this PR.

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java
Comment (Hronom) [reply modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java]: Done

**Comment by Hronom** [comment]
Also feel free to make adjustments on my behalf, you have my approve for this!

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java
Comment (reta) [reply modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java:85]: 👍 no objection from be, just feels a bit redundant

**Comment by Hronom** [comment]
Please check also this update to documentation https://github.com/opensearch-project/documentation-website/pull/500 I think it will be good to make more people know about it.

**Review by nknize — APPROVED** [review]
LGTM! Thank you for doing this!

**Review by kartg — COMMENTED** [review]
Late to the party, but 🚢

**Pull request merged**
